educe/stac/fusion.py

```diff
@@ -197,3 +201,47 @@ def speaker(self):
 # pylint: disable=invalid-name
 FakeRootEDU = _FakeRootEDU()
 # pylint: enable=invalid-name
+
+
+def fuse_edus(discourse_doc, unit_doc, postags):
+    """Return a copy of the discourse level doc, merging info
+    from both the discourse and units stage.
+
+    Any unit annotations which happen to be EDUs will be converted
+    to higher level EDUs
```

Comment:
Ah there's a bit of missing context which would help:

STAC has several kinds of unit-level annotations, of which

Structural annotations:

* paragraphs
* turns
* dialogues

EDUs: Segment, Offer, CounterOffer, etc

Resources, Preferences

So “units which happen to be EDUs” is talking about that choice
See educe.stac.annotation docs for typology

Will try to clarify this next week.


On Thu, 28 May 2015 at 16:12 Mathieu Morey <notifications@github.com> wrote:

> In educe/stac/fusion.py
> <https://github.com/irit-melodi/educe/pull/47#discussion_r31236387>:
>
> > @@ -197,3 +201,47 @@ def speaker(self):
> >  # pylint: disable=invalid-name
> >  FakeRootEDU = _FakeRootEDU()
> >  # pylint: enable=invalid-name
> > +
> > +
> > +def fuse_edus(discourse_doc, unit_doc, postags):
> > +    """Return a copy of the discourse level doc, merging info
> > +    from both the discourse and units stage.
> > +
> > +    Any unit annotations which happen to be EDUs will be converted
> > +    to higher level EDUs
>
> EDUs from both levels seem to coexist inside of the new EDU.
> If so, this last sentence is slightly confusing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/irit-melodi/educe/pull/47/files#r31236387>.
>